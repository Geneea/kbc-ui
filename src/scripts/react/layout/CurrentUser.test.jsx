import React from 'react';
import CurrentUser from './CurrentUser';
import { fromJS } from 'immutable';

const urlTemplates = fromJS({
  logout: '/admin/logout',
  maintainer: '/admin/maintainers/<%= maintainerId %>',
  changePassword: '/admin/change-password',
  manageApps: '/admin/manage-apps',
  syrupJobsMonitoring: '/admin/syrup-jobs-monitoring'
});

const user = fromJS({
  profileImageUrl: '/user.png',
  email: 'dev@keboola.com',
  name: 'dev user'
});

const maintainers = fromJS([
  {
    id: 1,
    name: 'First Maintainer'
  }
]);

describe('<CurrentUser />', function() {
  it('should render 40x40 icon, with email, no admin links, no maintainers, dropup false (no mode passed)', function() {
    shallowSnapshot(
      <CurrentUser
        user={user}
        maintainers={fromJS([])}
        urlTemplates={urlTemplates}
        canManageApps={false}
        dropup={true}
      />
    );
  });

  it('should render 20x20 icon, no email, no admin links, no maintainers, dropup true', function() {
    shallowSnapshot(
      <CurrentUser
        user={user}
        maintainers={fromJS([])}
        urlTemplates={urlTemplates}
        canManageApps={false}
        dropup={true}
        mode="single"
      />
    );
  });

  it('should render 20x20 icon, no email, no admin links, with 1 maintainer, dropup false', function() {
    shallowSnapshot(
      <CurrentUser
        user={user}
        maintainers={maintainers}
        urlTemplates={urlTemplates}
        canManageApps={false}
        dropup={false}
        mode="single"
      />
    );
  });

  it('should render 40x40 icon, with email, no admin links, no maintainers, dropup true', function() {
    shallowSnapshot(
      <CurrentUser
        user={user}
        maintainers={fromJS([])}
        urlTemplates={urlTemplates}
        canManageApps={false}
        dropup={true}
        mode="normal"
      />
    );
  });

  it('should render 40x40 icon, with email, with admin links, with 1 maintainer, dropup false', function() {
    shallowSnapshot(
      <CurrentUser
        user={user}
        maintainers={maintainers}
        urlTemplates={urlTemplates}
        canManageApps={true}
        dropup={false}
        mode="normal"
      />
    );
  });
});
