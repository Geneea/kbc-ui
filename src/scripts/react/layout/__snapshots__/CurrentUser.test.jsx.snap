exports[`<CurrentUser /> should render 20x20 icon, no email, no admin links, no maintainers, dropup true 1`] = `
<div
  className="kbc-user">
  <img
    className="kbc-user-avatar"
    height={20}
    src="/user.png"
    width={20} />
  <Uncontrolled(Dropdown)
    dropup={true}
    id="react-layout-current-user-dropdown"
    pullRight={true}>
    <DropdownToggle
      bsClass="dropdown-toggle"
      bsRole="toggle"
      bsStyle="link"
      noCaret={true}
      open={false}
      useAnchor={false}>
      <span
        className="kbc-icon-picker" />
      <span
        className="kbc-user-name">
        dev user
      </span>
    </DropdownToggle>
    <DropdownMenu
      bsClass="dropdown-menu"
      bsRole="menu"
      pullRight={false}>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/change-password">
        Account Settings
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={true}
        header={false} />
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/logout">
        Logout
      </MenuItem>
    </DropdownMenu>
  </Uncontrolled(Dropdown)>
</div>
`;

exports[`<CurrentUser /> should render 20x20 icon, no email, no admin links, with 1 maintainer, dropup false 1`] = `
<div
  className="kbc-user">
  <img
    className="kbc-user-avatar"
    height={20}
    src="/user.png"
    width={20} />
  <Uncontrolled(Dropdown)
    dropup={false}
    id="react-layout-current-user-dropdown"
    pullRight={true}>
    <DropdownToggle
      bsClass="dropdown-toggle"
      bsRole="toggle"
      bsStyle="link"
      noCaret={true}
      open={false}
      useAnchor={false}>
      <span
        className="kbc-icon-picker" />
      <span
        className="kbc-user-name">
        dev user
      </span>
    </DropdownToggle>
    <DropdownMenu
      bsClass="dropdown-menu"
      bsRole="menu"
      pullRight={false}>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/change-password">
        Account Settings
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={true}>
        Maintainers
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/maintainers/1">
        First Maintainer
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={true}
        header={false} />
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/logout">
        Logout
      </MenuItem>
    </DropdownMenu>
  </Uncontrolled(Dropdown)>
</div>
`;

exports[`<CurrentUser /> should render 40x40 icon, with email, no admin links, no maintainers, dropup false (no mode passed) 1`] = `
<div
  className="kbc-user">
  <img
    className="kbc-user-avatar"
    height={40}
    src="/user.png"
    width={40} />
  <Uncontrolled(Dropdown)
    dropup={true}
    id="react-layout-current-user-dropdown"
    pullRight={true}>
    <DropdownToggle
      bsClass="dropdown-toggle"
      bsRole="toggle"
      bsStyle="link"
      noCaret={true}
      open={false}
      useAnchor={false}>
      <span
        className="kbc-icon-picker" />
      <span
        className="kbc-user-name">
        dev user
      </span>
      <span
        className="kbc-user-email">
        dev@keboola.com
      </span>
    </DropdownToggle>
    <DropdownMenu
      bsClass="dropdown-menu"
      bsRole="menu"
      pullRight={false}>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/change-password">
        Account Settings
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={true}
        header={false} />
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/logout">
        Logout
      </MenuItem>
    </DropdownMenu>
  </Uncontrolled(Dropdown)>
</div>
`;

exports[`<CurrentUser /> should render 40x40 icon, with email, no admin links, no maintainers, dropup true 1`] = `
<div
  className="kbc-user">
  <img
    className="kbc-user-avatar"
    height={40}
    src="/user.png"
    width={40} />
  <Uncontrolled(Dropdown)
    dropup={true}
    id="react-layout-current-user-dropdown"
    pullRight={true}>
    <DropdownToggle
      bsClass="dropdown-toggle"
      bsRole="toggle"
      bsStyle="link"
      noCaret={true}
      open={false}
      useAnchor={false}>
      <span
        className="kbc-icon-picker" />
      <span
        className="kbc-user-name">
        dev user
      </span>
      <span
        className="kbc-user-email">
        dev@keboola.com
      </span>
    </DropdownToggle>
    <DropdownMenu
      bsClass="dropdown-menu"
      bsRole="menu"
      pullRight={false}>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/change-password">
        Account Settings
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={true}
        header={false} />
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/logout">
        Logout
      </MenuItem>
    </DropdownMenu>
  </Uncontrolled(Dropdown)>
</div>
`;

exports[`<CurrentUser /> should render 40x40 icon, with email, with admin links, with 1 maintainer, dropup false 1`] = `
<div
  className="kbc-user">
  <img
    className="kbc-user-avatar"
    height={40}
    src="/user.png"
    width={40} />
  <Uncontrolled(Dropdown)
    dropup={false}
    id="react-layout-current-user-dropdown"
    pullRight={true}>
    <DropdownToggle
      bsClass="dropdown-toggle"
      bsRole="toggle"
      bsStyle="link"
      noCaret={true}
      open={false}
      useAnchor={false}>
      <span
        className="kbc-icon-picker" />
      <span
        className="kbc-user-name">
        dev user
      </span>
      <span
        className="kbc-user-email">
        dev@keboola.com
      </span>
    </DropdownToggle>
    <DropdownMenu
      bsClass="dropdown-menu"
      bsRole="menu"
      pullRight={false}>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/change-password">
        Account Settings
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/manage-apps">
        Manage Applications
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/syrup-jobs-monitoring">
        Syrup Jobs Monitoring
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={true}>
        Maintainers
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/maintainers/1">
        First Maintainer
      </MenuItem>
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={true}
        header={false} />
      <MenuItem
        bsClass="dropdown"
        disabled={false}
        divider={false}
        header={false}
        href="/admin/logout">
        Logout
      </MenuItem>
    </DropdownMenu>
  </Uncontrolled(Dropdown)>
</div>
`;
